src/data/polynomial/degree/basic.lean

```diff
@@ -372,6 +423,43 @@ calc degree (p + q) = ((p + q).support).sup some : rfl
 lemma leading_coeff_eq_zero_iff_deg_eq_bot : leading_coeff p = 0 ↔ degree p = ⊥ :=
 by rw [leading_coeff_eq_zero, degree_eq_bot]
 
+lemma nat_degree_mem_support_of_nonzero (H : p ≠ 0) : p.nat_degree ∈ p.support :=
+(p.mem_support_to_fun p.nat_degree).mpr ((not_congr leading_coeff_eq_zero).mpr H)
+
+lemma nat_degree_eq_support_max' (h : p ≠ 0) :
+  p.nat_degree = p.support.max' (nonempty_support_iff.mpr h) :=
+begin
+  apply le_antisymm,
+  { apply finset.le_max',
+    rw mem_support_iff_coeff_ne_zero,
+    exact (not_congr leading_coeff_eq_zero).mpr h, },
+  { apply max'_le,
+    refine le_nat_degree_of_mem_supp, },
+end
+
+lemma nat_degree_C_mul_X_pow_le (a : R) (n : ℕ) : nat_degree (C a * X ^ n) ≤ n :=
+begin
+  by_cases a0 : a = 0,
+    { rw [a0, C_0, zero_mul, nat_degree_zero],
+      exact nat.zero_le _, },
+    { rw nat_degree_eq_support_max',
+      { simp_rw [support_C_mul_X_pow_nonzero a0, max'_singleton n], },
+      { intro,
+        apply a0,
+        rw [← C_inj, C_0],
+        apply mul_X_pow_eq_zero a_1, }, },
+end
+
+lemma nat_degree_C_mul_X_pow_nonzero {a : R} (n : ℕ) (ha : a ≠ 0) : nat_degree (C a * X ^ n) = n :=
+begin
+  rw nat_degree_eq_support_max',
+    { simp_rw [support_C_mul_X_pow_nonzero ha, max'_singleton n], },
```

Comment:
Formatting.

Reply:
Hi @jcommelin , thank you very much for your comments!  I just do not understand what is wrong with the formatting here...

Reply:
You are indenting too much :wink: 
The `{` should be on the same level as the line above it.

Reply:
Ah, i get it now!  I actually thought that this was the correct formatting!  I will change it, thanks!

Reply:
The other one looks good now, but this one still needs updating :wink:

Reply:
I think that I took care of all the formatting issues!